src/status_im2/contexts/chat/composer/view.cljs

```diff
@@ -62,50 +63,54 @@
+     [mentions/view props state animations max-height cursor-pos
+      (:images subs)
+      (:link-previews? subs)
+      (:reply subs)
       (:edit subs)]
-     [gesture/gesture-detector
-      {:gesture (drag-gesture/drag-gesture props state animations dimensions keyboard-shown)}
-      [reanimated/view
-       {:style     (style/sheet-container insets state animations)
-        :on-layout #(handler/layout % state blur-height)}
-       [sub-view/bar]
-       [reply/view state]
-       [edit/view state]
-       [reanimated/touchable-opacity
-        {:active-opacity      1
-         :on-press            (when @(:input-ref props) #(.focus ^js @(:input-ref props)))
-         :style               (style/input-container (:height animations) max-height)
-         :accessibility-label :message-input-container}
-        [rn/selectable-text-input
-         {:ref        #(reset! (:selectable-input-ref props) %)
-          :menu-items @(:menu-items state)
-          :style      (style/input-view state)}
-         [rn/text-input
-          {:ref                      #(reset! (:input-ref props) %)
-           :default-value            @(:text-value state)
-           :on-focus                 #(handler/focus props state animations dimensions)
-           :on-blur                  #(handler/blur state animations dimensions subs)
-           :on-content-size-change   #(handler/content-size-change %
-                                                                   state
-                                                                   animations
-                                                                   dimensions
-                                                                   (or keyboard-shown (:edit subs)))
-           :on-scroll                #(handler/scroll % props state animations dimensions)
-           :on-change-text           #(handler/change-text % props state)
-           :on-selection-change      #(handler/selection-change % props state)
-           :on-selection             #(selection/on-selection % props state)
-           :max-height               max-height
-           :max-font-size-multiplier 1
-           :multiline                true
-           :placeholder              (i18n/label :t/type-something)
-           :placeholder-text-color   (colors/theme-colors colors/neutral-40 colors/neutral-50)
-           :style                    (style/input-text props state)
-           :max-length               constants/max-text-size
-           :accessibility-label      :chat-message-input}]]
-        [gradients/view props state animations show-bottom-gradient?]]
-       [images/images-list]
-       [actions/view props state animations window-height insets subs]]]]))
+     [reanimated/view
+      {:style     (style/sheet-container insets state animations)
+       :on-layout #(handler/layout % state blur-height)}
+      [gesture/gesture-detector
```

Comment:
FYI: The critical change here is that I moved the gesture to surround only the `sub-view/bar` component, otherwise it was stealing the horizontal scrolling events from the link preview FlatList.

Reply:
I think this list is similar to images list. It should work without that change.

Reply:
Done. I reverted back to the original implementation and used the `gesture/flat-list` component as you suggested.